mongo/client.go

```diff
@@ -63,6 +63,7 @@ type Client struct {
 	writeConcern   *writeconcern.WriteConcern
 	registry       *bsoncodec.Registry
 	monitor        *event.CommandMonitor
+	interceptor    *event.CommandInterceptor
```

Comment:
With this sort of approach it looks like we'd need to ensure that the interceptor actually gets passed down into the various `Operation`s. I didn't do that as part of this draft just to keep it small/timeboxed as an example.